### DISCUSSION TITLE
newly installed programs not on path

### DISCUSSION THREAD

**Discussion by ja-gooding** [Homebrew/discussions#2632] [Everyday usage] [answered]
Whenever I install a new program with brew à la `brew install foo`, `foo` is not immediately available on the path unless I reload the terminal session, or open a new one. I don't remember this previously being an issue?

Details:
macOS 12.1, apple M1
```bash
╰─ brew -v                                                                                                                                                                    
Homebrew 3.3.8-6-g29426f0
Homebrew/homebrew-core (git revision 1039088316e; last commit 2021-12-14)
Homebrew/homebrew-cask (git revision 947f995586; last commit 2021-12-14)
```
zsh, with oh-my-zsh

**Comment by SMillerDev** [comment]
That's likely some caching in your zsh setup. Nothing brew can do about it

**Reply by ja-gooding** [reply]
Do you know where I should look in zsh for this? What I should even be searching for?

**Reply by SMillerDev** [reply]
https://github.com/ohmyzsh/ohmyzsh/issues/3440 would be good to read

**Reply by ja-gooding** [reply]
https://github.com/Homebrew/brew/issues/2202 I know this is a little old, but I can't find anything more recent so I assume this is still the case, but he refers to wrapping home-brew so that you could run post install scripts, (like in my case I want to automatically run `rehash` after an installation / upgrade / remove action. But I have no idea what he means by that, and I can't think of a way off the top of my head that would allow you preserve tab completion for home-brew.

**Comment by ja-gooding** [comment, marked as answer]
> [ohmyzsh/ohmyzsh#3440](https://github.com/ohmyzsh/ohmyzsh/issues/3440) would be good to read

Thanks to @SMillerDev for finding this. I confused myself a little. It turns out they are on the path, but I am a prolific user of tab completion, and it was tab completion that wasn't working. Running `rehash` for zsh fixes the issue, the linked issue gives an option for continuous rehashing, however that is not necessarily appropriate for everyone depending on their drive setups.
